src/createReducer.js

```diff
@@ -361,6 +361,11 @@ function createReducer<M, L>(structure: Structure<M, L>) {
         result = setIn(result, 'values', values)
         result = setIn(result, 'initial', values)
       }
+      var submitFailed = getIn(state, 'submitFailed');
```

Comment:
Yes, I don't agree with that.
Perhaps @slim25's idea is that every form start as invalid - which is not true, because this state depends on the validation functions.